src/tsdb.c

```diff
@@ -604,4 +605,37 @@ int SeriesCalcRange(Series *series,
 
 timestamp_t CalcWindowStart(timestamp_t timestamp, size_t window) {
     return timestamp - (timestamp % window);
+}
+
+timestamp_t SkipExpiredSamples(Series *series, long long *skipped) {
+    *skipped = 0;
+    if (series->totalSamples == 0) {
+        return 0;
+    }
+
+    size_t i = 0;
+    Chunk_t *chunk;
+    bool rev = false;
+    Sample sample = { 0 };
+    ChunkFuncs *funcs = series->funcs;
+    timestamp_t minTimestamp =
+        series->retentionTime ? series->lastTimestamp - series->retentionTime : 0;
+
+    SeriesTrim(series);
+    RedisModuleDictIter *iter = RedisModule_DictIteratorStartC(series->chunks, "^", NULL, 0);
```

Comment:
Why not use the series iterator?

Reply:
Since we need to know how many samples were skipped and we cannot get it from the `SeriesIterator`.